lib/record_store/zone/config/ignore_pattern.rb

```diff
@@ -0,0 +1,36 @@
+module RecordStore
+  class Zone
+    class Config
+      class IgnorePattern
+        attr_accessor :orig_hash
+        alias_method :to_hash, :orig_hash
+
+        def initialize(orig_hash)
+          @orig_hash = orig_hash
+        end
+
+        def should_ignore?(record)
+          all_pairs.all?{|(key, value)| record.respond_to?(key) && value_matches?(value, record.send(key))}
+        end
+
+        def all_pairs
+          orig_hash.reject{|key, value| key === 'match'}
+        end
+
+        def value_matches?(ignore_pattern_value, record_value)
+          return record_value.match?(ignore_pattern_value) if is_regex?
+          return record_value === ignore_pattern_value if is_exact?
+          false
+        end
+
+        def is_regex?
+          orig_hash['match'] === 'regex'
+        end
+
+        def is_exact?
+          !orig_hash.key?('match') || (orig_hash['match'] === 'exact')
```

Comment:
`origin_hash['match'] == "exact"` is more than enough here, right? Ruby will return `nil` if you there's no key named `match` and that will evaluate the condition to false.